pkg/fileservice/counter.go

```diff
@@ -57,3 +65,23 @@ func WithCounter(ctx context.Context, counter *Counter) context.Context {
 	newCounters = append(newCounters, counter)
 	return context.WithValue(ctx, CtxKeyCounters, newCounters)
 }
+
+func (c *Counter) Collect() metric.Stats {
+	stats := make(map[string]int64)
+
+	c.Lock()
```

Comment:
why a lock is useful when writer does not lock?

Reply:
I am assuming, we need to write a snapshot of all the Counter values.

Reply:
While Collecting, the initial state is 
```
NumRead = 5
NumHit = 5
```
We collected `NumRead=5`

Before collecting `NumHit=5`, `NumHit` is changed to `NumHit=7`

Now we collect `NumHit = 7`.

Should the output in the log be 
```
NumRead = 5
NumHit = 5
```
Or
```
NumRead = 5
NumHit = 7
```